akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -2193,15 +2337,19 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
         replyTo ! g
       }
     }
-    val myMissingKeys = otherKeys.diff(myKeys)
+    val myMissingKeys =
+      if (expiryEnabled)
+        otherKeys.diff(myKeys).filterNot(key =>isExpired(key))
+      else
+        otherKeys.diff(myKeys)
```

Comment:
Would it make sense to filter myKeys and otherKeys on expiry up front instead perhaps, instead ad-hoc filtering, or do we need all later below?